core/src/main/java/com/oath/oak/OakUnsafeRef.java

```diff
@@ -0,0 +1,28 @@
+package com.oath.oak;
+
+import java.nio.ByteBuffer;
+import java.util.function.Function;
+
+public interface OakUnsafeRef {
+
+    /**
+     * @return the underlying ByteBuffer at position 0
+     */
+    ByteBuffer getByteBuffer();
+
+    /**
+     * @return the key/value offset in the underlying ByteBuffer
+     */
+    int getOffset();
+
+    /**
+     * @return the key/value length
+     */
+    int getLength();
+
+    /**
+     * @return the exact address of the underlying buffer in the position of the key/value
+     */
+    long getAddress();
+
```

Comment:
It should have transform() method as well.

Reply:
Note that at this point, `OakBuffer`s `transform()` method acts on a `ByteBuffer`, so adding a `transform()` method here does not make much sense.
In addition, thinking forward, wouldn't it be easier to just call the `transform()` method of the `OakBuffer` and then cast the input buffer to `OakUnsafeRef` instead of having to deal with two different kinds of transform method types?

Reply:
1. I understand that currently you can manage without transform here, but if already adding this "OakUnsafeDirectBuffer" I believe it should be the full interface.
2. Transform with safe/unsafe buffer is much more comfortable. If you do the cast once you can work with your new UnsafeBuffer only. Otherwise you need to jungle between Safe/Unsafe buffers.

Reply:
In general, agree with both statements. 
1. It's better to provide one consistent way of doing things. 
2. However, the Oak{R,W}Buffer world must remain self-sufficient. If a transform() there is not just a syntactic sugar, we should support it too. That might be in conflict with (1).

Reply:
Adding a transform here is syntactic sugar.

Instead of writing:
```java
OakRBuffer b = ...;
b.transform(rb -> {
    OakUnsafeDirectBuffer urb = (OakUnsafeDirectBuffer) rb;
    long address = urb.getAddress();
   ...
});
```
You'll be able to write:
```java
OakRBuffer b = ...;
OakUnsafeDirectBuffer ub = (OakUnsafeDirectBuffer) b;
ub.transform(urb -> {
    long address = urb.getAddress();
   ...
});
```
* Notice that the casting was moved outside of the transform lambda expression.

I think this API should remain separate from the buffer API that is used for regular access.
That is, its only purpose should be getting the underlying data.
Any "Oak" related logic and interface should be used from the regular buffer interface.